### PR TITLE
Add auto-completion document for karmadactl

### PR DESCRIPTION
**Pull request by lonelyCZ**
Signed-off-by: lonelyCZ <531187475@qq.com>

**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
Add a document about using auto-completion of `karmadactl`

**Which issue(s) this PR fixes**:
Part #1046 

**Special notes for your reviewer**:
I consider firstly introduce the auto-completion document of `karmadactl`, if there will be official support for `plugin` auto-completion, we'll supplement the document about `kubectl karmada`.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by RainbowMango — COMMENTED** [review]
So sorry that I forgot to submit my comments on it.

File: docs/bash-auto-completion-on-linux.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# bash auto-completion on Linux
+You can enable karmadactl autocompletion in any of the following ways:
+- Install bash-completion
+- Enable karmadactl autocompletion
+
+> you can test if you have bash-completion already installed by running `type _init_completion`
```
[the remark this answers is not in the document]
Comment (lonelyCZ) [reply docs/bash-auto-completion-on-linux.md:6]: Ok, directly reference it?

[the remark this answers is not in the document]
Comment (RainbowMango) [reply docs/bash-auto-completion-on-linux.md:6]: That's totally doable.

[the remark this answers is not in the document]
Comment (lonelyCZ) [reply docs/bash-auto-completion-on-linux.md]: done

**Pull request merged**